Tests/IGListAdapterTests.m

```diff
@@ -1067,6 +1067,24 @@ - (void)test_whenSupplementarySourceReturnsNegativeSize_thatAdapterReturnsZero {
     XCTAssertEqual(size.height, 0.0);
 }
 
+- (void)test_whenSectionIndexGreaterThanZero_thatAdapterReturnsSizeForSectionIndexesGreaterThanZero {
+  self.dataSource.objects = @[@1];
+  [self.adapter reloadDataWithCompletion:nil];
+  
+  IGTestSupplementarySource *supplementarySource = [IGTestSupplementarySource new];
+  supplementarySource.collectionContext = self.adapter;
+  supplementarySource.supportedElementKinds = @[UICollectionElementKindSectionFooter];
+  
+  IGListSectionController *controller = [self.adapter sectionControllerForObject:@1];
+  controller.supplementaryViewSource = supplementarySource;
+  supplementarySource.sectionController = controller;
+  
+  const CGSize size = [self.adapter sizeForSupplementaryViewOfKind:UICollectionElementKindSectionHeader
+                                                       atIndexPath:[NSIndexPath indexPathForItem:0 inSection:1]];
```

Comment:
> You want every section to have an index 0 because each section controller is some sort of auto-sufficient unit that has no knowledge about the other sections

💯 

> I thought this was possible and intended to be possible from the index parameter that is being passed to the supplementary view source.

You're right that it's not possible when a completely separate object implements this protocol. We originally designed this API thinking section controllers would be the ones implementing `IGListSupplementaryViewSource` and not separate objects. Since then, internally we have implementations creating their own supplementary source objects but also providing refs to a section controller as a workaround.

Atm a section controller can get info about its position with [these APIs](https://github.com/Instagram/IGListKit/blob/master/Source/IGListSectionController.h#L100-L113), so if the section controller implemented `IGListSupplementaryViewSource`, it becomes trivial to get the section.

> what should I do in order to be able to receive different indexes in my IGListSupplementaryViewSource implementation?

I'd make your section controllers conform to the protocol and use `self.isFirstSection` to determine if its the first section controller or not.

> let's say that I succeed to have 2 supplementary views, which are the same class but they have different sizes. So I have indexes 0 and 1 for my supplementary views. How can I code the view index 0 to be used for section index 0, and the view index 1 to be used for the other section indexes?

If I'm understanding correctly, I think that might overcomplicate things. Instead do something like this:

```swift
class MySectionController: IGListSupplementaryViewSource {

  // implement stuff you need here

  // MARK: IGListSupplementaryViewSource

  func supportedElementKinds() -> [String] {
    return [UICollectionElementKindSectionHeader]
  }

  func viewForSupplementaryElement(ofKind elementKind: String, at index: Int) -> UICollectionReusableView {
  let view = collectionContext?.dequeueReusableSupplementaryView(
    ofKind: UICollectionElementKindSectionHeader,
    for: self,
    class: MyViewClass.self,
    at: index) as! MyViewClass
  // do some config
  return view
  }

  func sizeForSupplementaryView(ofKind elementKind: String, at index: Int) -> CGSize {
    let width = collectionContext!.containerSize.width
    let height: CGFloat = isFirstSection ? 100 : 50
    return CGSize(width: width, height: height)
  }

}
```